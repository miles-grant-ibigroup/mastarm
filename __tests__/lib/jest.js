// @flow

const jestUtils = require('../../lib/jest')

const JEST_CONFIG_INDEX = 4

describe('jest.js', () => {
  it('generateTestConfig should generate proper config', () => {
    const cfg = jestUtils.generateTestConfig(['these', 'files', 'only'], {
      cache: false,
      coveragePaths: 'bin src another-folder',
      customConfigFile: '__tests__/test-utils/mocks/mock-jest-config.json',
      runInBand: true,
      setupFiles: 'beforeTestsSetup.js',
      testEnvironment: 'node',
      updateSnapshots: true
    })
    cfg[JEST_CONFIG_INDEX] = JSON.parse(cfg[JEST_CONFIG_INDEX])
    expect(cfg[JEST_CONFIG_INDEX].transform['^.+\\.(t|j)sx?$']).toContain(
      'lib/jest-preprocessor.js'
    )
    delete cfg[JEST_CONFIG_INDEX].transform
    expect(cfg).toMatchSnapshot()
  })

  it('should be able to load a file that requires yaml', () => {
    const indexWithYaml = require('../test-utils/mocks/index-with-yaml')
    expect(indexWithYaml).toBeTruthy()
  })

  it('should be able to perform an async test', async () => {
    await Promise.resolve()
  })
})
