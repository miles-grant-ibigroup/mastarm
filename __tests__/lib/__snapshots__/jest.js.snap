// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest.js generateTestConfig should generate proper config 1`] = `
Array [
  "--updateSnapshot",
  "--no-cache",
  "--runInBand",
  "--config",
  Object {
    "collectCoverage": true,
    "collectCoverageFrom": Array [
      "lib/**/*.js",
      "bin",
      "src",
      "another-folder",
    ],
    "coverageDirectory": "coverage",
    "moduleFileExtensions": Array [
      "js",
      "ts",
      "jsx",
      "json",
      "png",
      "yml",
    ],
    "notify": true,
    "setupFiles": Array [
      "beforeTestsSetup.js",
    ],
    "testEnvironment": "node",
    "testPathIgnorePatterns": Array [
      "<rootDir>/node_modules/",
      "<rootDir>/__tests__/test-utils",
    ],
    "testURL": "http://localhost:9966",
  },
  "these",
  "files",
  "only",
]
`;
